enterprise/server/backends/s3_cache/s3_cache.go

```diff
@@ -107,17 +108,30 @@ func NewS3Cache(awsConfig *config.S3CacheConfig) (*S3Cache, error) {
 	return s3c, nil
 }
 
+func (s3c *S3Cache) bucketExists(ctx context.Context, bucketName string) (bool, error) {
+	ctx, spn := tracing.StartSpan(ctx)
+	_, err := s3c.s3.HeadBucketWithContext(ctx, &s3.HeadBucketInput{Bucket: aws.String(bucketName)})
+	spn.End()
+	if err == nil {
+		return true, nil
+	}
```

Comment:
nit: this might be clearer with a defer?

    ctx, spn := tracing.StartSpan(ctx)
    defer spn.End()
	if _, err := s3c.s3.HeadBucketWithContext(ctx, &s3.HeadBucketInput{Bucket: aws.String(bucketName)}); err == nil {
		return true, nil
	}
    ...

Reply:
Done.